diff -ruN -x '*.orig' -x '*.rej' ../src/x.c ./x.c
--- ../src/x.c	2026-06-19 14:34:02.727203628 -0700
+++ ./x.c	2026-06-19 15:25:43.921024140 -0700
@@ -1467,6 +1467,11 @@
 	if (base.mode & ATTR_INVISIBLE)
 		fg = bg;
 
+	// gradient
+	bg->color.alpha = grad_alpha * 0xffff * (win.h - y*win.ch) / win.h + stat_alpha * 0xffff;
+	// uncomment to invert the gradient
+	// bg->color.alpha = grad_alpha * 0xffff * (y*win.ch) / win.h + stat_alpha * 0xffff;
+
 	/* Intelligent cleaning up of the borders. */
 	if (x == 0) {
 		xclear(0, (y == 0)? 0 : winy, borderpx,
